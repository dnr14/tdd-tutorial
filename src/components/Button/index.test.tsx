import { fireEvent, render, screen } from '@testing-library/react';
import { Button } from '.';
import 'jest-styled-components';

describe('<Button>', () => {
  it('<Button> 랜더링', () => {
    const { container } = render(<Button label="추가" />);

    const label = screen.getByText('추가');
    expect(label).toBeInTheDocument();
    expect(container).toMatchSnapshot();
    expect(container).toMatchSnapshot();
  });
  it('<Button> background, hoverColor change Test', () => {
    const backgroundColor = '#304ffe';
    const hoverColor = '#1e40ff';
    render(<Button label="추가" backgroundColor={backgroundColor} hoverColor={hoverColor} />);

    const parent = screen.getByTestId('button-container');
    expect(parent).toHaveStyleRule('background-color', '#304ffe');
    expect(parent).toHaveStyleRule('background-color', '#1e40ff', {
      modifier: ':hover',
    });
    expect(parent).toHaveStyleRule('box-shadow', '5px 5px 10px rgba(0,0,0,0.2)', {
      modifier: ':active',
    });
  });

  it('<Button> click Test', () => {
    const handleClick = jest.fn();
    render(<Button label="추가" onClick={handleClick} />);
    const label = screen.getByText('추가');
    expect(label).toBeInTheDocument();
    expect(handleClick).toHaveBeenCalledTimes(0);
    fireEvent.click(label);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
