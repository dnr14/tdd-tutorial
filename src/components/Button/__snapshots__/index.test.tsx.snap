// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> <Button> 랜더링 1`] = `
.c0 {
  text-align: center;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  background-color: #304ffe;
}

.c0:hover {
  background-color: #1e40ff;
}

.c0:active {
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

.c1 {
  color: #fff;
}

<div>
  <div
    class="c0"
    data-testid="button-container"
  >
    <div
      class="c1"
    >
      추가
    </div>
  </div>
</div>
`;

exports[`<Button> <Button> 랜더링 2`] = `
.c0 {
  text-align: center;
  padding: 10px 20px;
  border-radius: 8px;
  cursor: pointer;
  background-color: #304ffe;
}

.c0:hover {
  background-color: #1e40ff;
}

.c0:active {
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

.c1 {
  color: #fff;
}

<div>
  <div
    class="c0"
    data-testid="button-container"
  >
    <div
      class="c1"
    >
      추가
    </div>
  </div>
</div>
`;
